pkg/code-generator/collector/collector.go

```diff
@@ -0,0 +1,284 @@
+package collector
+
+import (
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
+	"sync"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/gogo/protobuf/protoc-gen-gogo/descriptor"
+	"github.com/solo-io/go-utils/log"
+	"github.com/solo-io/go-utils/stringutils"
+	"github.com/solo-io/solo-kit/pkg/code-generator/model"
+	"github.com/solo-io/solo-kit/pkg/code-generator/parser"
+	"github.com/solo-io/solo-kit/pkg/errors"
+	"golang.org/x/sync/errgroup"
+)
+
+type Collector interface {
+	CollectDescriptorsFromRoot(root string, skipDirs []string) ([]*model.DescriptorWithPath, error)
```

Comment:
please comment this interface. what does it do? how does it work?

Reply:
It does nothing differently from before, I made it an interface only so we could possibly add tests at a later date